### PR TITLE
Add the `tribe_get_organizer_object` function.

### PR DESCRIPTION
**Pull request by lucatume**
Issue: n/a

This PR adds the `tribe_get_organizer_object` function and, 
with it, all the attached structure to return a list of
Organizer post objects on `tribe_get_event()->organizers`.
The previous return value of the method, a collection of
Organizer names, is moved to the `tribe_get_event()->organizer_names`
method.

Artifacts: see tests.

### CODE REVIEW

**Review by Camwyn — DISMISSED** [review]
Looks legit 😉

**Review by jesseeproductions — APPROVED** [review]
Looks good.

File: tests/wpunit/functions/template-tags/organizerTest.php [hunk 1]
```diff
@@ -85,4 +86,86 @@ public function test_tribe_sanitize_organizers() {
...
+
+	/**
+	 * It should allow getting an organizer decorated post object
+	 *
+	 * @test
+	 */
+	public function should_allow_getting_an_organizer_decorated_post_object() {
+		$this->assertNull( tribe_get_organizer_object( PHP_INT_MAX ) );
+
+		$organizer_id       = $this->given_an_organizer();
```
Comment (paulmskim) [inline tests/wpunit/functions/template-tags/organizerTest.php:98]:
```suggestion
		$organizer_id = $this->given_an_organizer();
```

File: tests/wpunit/functions/template-tags/organizerTest.php [hunk 2]
```diff
@@ -85,4 +86,86 @@ public function test_tribe_sanitize_organizers() {
...
+	 * It should return the data of the global organizer when set
+	 *
+	 * @test
+	 */
+	public function should_return_the_data_of_the_global_organizer_when_set() {
+		$organizer_id       = $this->given_an_organizer();
+		$GLOBALS['post'] = get_post($organizer_id);
+		$GLOBALS['post_id'] = $organizer_id;
+
+		$organizer    = tribe_get_organizer_object( null );
```
Comment (paulmskim) [inline tests/wpunit/functions/template-tags/organizerTest.php:118]:
```suggestion
		$organizer_id       = $this->given_an_organizer();
		$GLOBALS['post']    = get_post($organizer_id);
		$GLOBALS['post_id'] = $organizer_id;

		$organizer = tribe_get_organizer_object( null );
```

File: tests/wpunit/functions/template-tags/organizerTest.php [hunk 3]
```diff
@@ -85,4 +86,86 @@ public function test_tribe_sanitize_organizers() {
...
+
+	/**
+	 * It should allow getting the organizer as associative_array
+	 *
+	 * @test
+	 */
+	public function should_allow_getting_the_organizer_as_associative_array() {
+		$organizer_id       = $this->given_an_organizer();
+
+		$organizer    = tribe_get_organizer_object( $organizer_id, ARRAY_A );
```
Comment (paulmskim) [inline tests/wpunit/functions/template-tags/organizerTest.php:135]:
```suggestion
		$organizer_id = $this->given_an_organizer();

		$organizer = tribe_get_organizer_object( $organizer_id, ARRAY_A );
```

File: src/Tribe/Models/Post_Types/Event.php [hunk 4]
```diff
@@ -206,8 +207,15 @@ static function () use ( $post_id ) {
...
+					new Lazy_Post_Collection(
+						$organizer_fetch,
+						'tribe_get_organizer_object'
+					)
+				)->on_resolve( $cache_this ),
+				'venues'                 => (
+					new Lazy_Post_Collection(
 					$venue_fetch,
 					'tribe_get_venue_object' )
 				)->on_resolve( $cache_this ),
```
Comment (paulmskim) [inline src/Tribe/Models/Post_Types/Event.php:221]: got a bit of weird spacing here

**Review by paulmskim — COMMENTED** [review]
just a few small things

File: src/Tribe/Models/Post_Types/Organizer.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	 * {@inheritDoc}
+	 *
+	 * @since TBD
+	 */
+	protected function get_cache_slug() {
+		return 'organizers';
+	}
+
+	/**
+	 * {@inheritDoc}
```
Comment (tr1b0t) [inline src/Tribe/Models/Post_Types/Organizer.php:33]: :no_entry_sign: Doc comment for parameter "$filter" missing

<sub>hash: c3c1df14766413edd69abad18e1c7968</sub>

File: src/Tribe/Organizer.php [hunk 6]
```diff
@@ -647,12 +648,51 @@ public static function get_fetch_callback( $event ) {
...
-				? array_map( 'tribe_get_organizer', $organizer_ids )
+			$organizers = ! empty( $organizer_ids )
+				? array_map( 'tribe_get_organizer_object', $organizer_ids )
 				: [];
 
 			return array_filter( $organizers );
 		};
 	}
 
+	/**
```
Comment (tr1b0t) [inline src/Tribe/Organizer.php:659]: :no_entry_sign: Doc comment for parameter "$event" missing

<sub>hash: c99bc1e6642ad11d16717c28b9dbb6e3</sub>

File: src/functions/template-tags/organizer.php [hunk 7]
```diff
@@ -541,4 +543,85 @@ function tribe_get_organizers( $only_with_upcoming = false, $posts_per_page = -
...
+		 */
+		$return = apply_filters( 'tribe_get_organizer_object_before', null, $organizer, $output, $filter );
+
+		if ( null !== $return ) {
+			return $return;
+		}
+
+		$post = false;
+		if ( ! $force ) {
+			$cache_key = 'tribe_get_organizer_object_' . md5( json_encode( [ $organizer, $output, $filter ] ) );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:590]: :warning: json_encode() is discouraged. Use wp_json_encode() instead.

<sub>hash: af79d9155ee022613e435a8d421b5fa8</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)
:warning: 1 warning(s)

**Pull request merged**